common/cards/default/hermits/goodtimeswithscar-rare.ts

```diff
@@ -97,13 +89,16 @@ class GoodTimesWithScarRareHermitCard extends HermitCard {
 
 				opponentPlayer.hooks.afterAttack.remove(instance)
 				player.hooks.onTurnStart.remove(instance)
-				if (canCleanUp()) player.hooks.onHermitDeath.remove(instance)
+				if (canCleanUp()) player.hooks.afterAttack.remove(instance)
 			})
-			player.hooks.onHermitDeath.add(instance, (hermitPos) => {
-				const hermitInstance = hermitPos.card?.cardInstance
-				if (hermitInstance) delete canRevives[hermitInstance]
-
-				if (canCleanUp()) player.hooks.onHermitDeath.remove(instance)
+			player.hooks.afterAttack.add(instance, (attack) => {
```

Comment:
```suggestion
			player.hooks.afterDefence.add(instance, (attack) => {
```
Please add a similar hook in onAttach so targetInstance is deleted when your hermit dies from any cause and after Totem has a chance to revive the row